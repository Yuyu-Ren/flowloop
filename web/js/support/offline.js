// Offline plugin
import * as OfflinePlugin from 'offline-plugin/runtime'
OfflinePlugin.install()
