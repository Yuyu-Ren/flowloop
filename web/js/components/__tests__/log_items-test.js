/* eslint-env jest */

import LogItems from '../log_items'

import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { labels, logs } from '../../../../test/support/fixtures'

describe('LogItems', () => {
  it('exists', () => {
    expect(LogItems).toBeTruthy()
  })

  it('renders', () => {
    const comp = <LogItems items={logs} labels={labels} />
    const wrapper = shallow(comp)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
