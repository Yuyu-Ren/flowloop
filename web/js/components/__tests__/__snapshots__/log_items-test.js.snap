// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogItems renders 1`] = `
<span
  className="timeline-small-items"
>
  <LogItem
    item={
      Object {
        "duration": 1800000,
        "endedAt": 2010-04-20T10:30:00.000Z,
        "id": "_log0",
        "isComplete": true,
        "labelId": "_work",
        "startedAt": 2010-04-20T10:00:00.000Z,
        "timerType": "work",
      }
    }
    labels={
      Object {
        "_chore": Object {
          "color": "red",
          "id": "_chore",
          "name": "Chore",
        },
        "_work": Object {
          "color": "blue",
          "id": "_work",
          "name": "Work",
        },
      }
    }
  />
  <LogItem
    item={
      Object {
        "duration": 1800000,
        "endedAt": 2010-04-20T11:30:00.000Z,
        "id": "_log1",
        "isComplete": true,
        "labelId": "_chore",
        "startedAt": 2010-04-20T11:00:00.000Z,
        "timerType": "work",
      }
    }
    labels={
      Object {
        "_chore": Object {
          "color": "red",
          "id": "_chore",
          "name": "Chore",
        },
        "_work": Object {
          "color": "blue",
          "id": "_work",
          "name": "Work",
        },
      }
    }
  />
</span>
`;
